src/utils/signaling.js

```diff
@@ -141,6 +141,22 @@ Signaling.Base.prototype._trigger = function(ev, args) {
 	EventBus.$emit('signaling-' + kebabCase(ev), args)
 }
 
+Signaling.Base.prototype.setSettings = function(settings) {
+	if (!settings) {
+		// Signaling object is expected to always have a settings object
+		return
+	}
+
+	this.settings = settings
+
+	if (this._pendingUpdateSettingsPromise) {
+		const pendingUpdateSettingsPromise = this._pendingUpdateSettingsPromise
+		delete this._pendingUpdateSettingsPromise
+
+		pendingUpdateSettingsPromise.resolve()
```

Comment:
```suggestion
		this._pendingUpdateSettingsPromise.resolve()
		delete this._pendingUpdateSettingsPromise
```
We have a promise and assigned resolve to it at this step, so suggestion should work as well (tested)